plugin.rb

```diff
@@ -137,6 +137,31 @@ def self.allowed_group_ids
     results
   end
 
+  if respond_to?(:register_upload_unused_callback)
+    register_upload_unused_callback do |uploads|
+      uploads
+        .joins("LEFT JOIN chat_uploads pu ON cu.upload_id = uploads.id")
+        .where("cu.upload_id IS NULL")
+    end
+  end
+
+  if respond_to?(:register_upload_in_use)
+    register_upload_in_use do |upload|
+      # TODO after May 2022 - remove this. No longer needed as chat uploads are in a table
+      next true if ChatMessage.where("message LIKE ? OR message LIKE ?", "%#{upload.sha1}%", "%#{encoded_sha}%").exists?
```

Comment:
Where is `encoded_sha` defined?

Reply:
I replaced that with an upload property.